### PR TITLE
add expiration date field to VerificationDocument

### PR DESCRIPTION
**Pull request by rcabrera-py**
## Summary by CodeRabbit

* **New Features**
  * Verification documents now expose parsed expiration and emission date properties (return date objects when available, otherwise None).

* **Public API**
  * Verification document type is now publicly available for use.

* **Tests**
  * Added tests covering expiration and emission date behavior for present, missing, and None cases.

* **Chores**
  * Bumped package version to a new release.

### CODE REVIEW

File: mati/types/enums.py [hunk 1]
```diff
@@ -184,6 +184,26 @@ def ocr_number(self) -> str:
             return self.fields['ocr_number']['value']
         return ''
 
+    @property
+    def expiration_date(self) -> str:
```
Comment (felipao-mx) [inline mati/types/enums.py]: hacer que estas nuevas properties retornen `datetime.date`  o. `None` directo en vez de `str`

**Review by felipao-mx — CHANGES_REQUESTED** [review]

File: mati/types/enums.py [hunk 2]
```diff
@@ -184,6 +184,16 @@ def ocr_number(self) -> str:
...
+        """
+        This property fills the expiration date direct from the ocr
+        fields `expiration_date`
+        """
+        if self.fields and 'expiration_date' in self.fields:
+            return self.fields['expiration_date']['value']
+        return ''
```
[the remark this answers is not in the document]
Comment (felipao-mx) [reply mati/types/enums.py]: @rcabrera-py atender este comentario de @coderabbitai

[the remark this answers is not in the document]
Comment (rcabrera-py) [reply mati/types/enums.py]: done

File: tests/test_types.py [hunk 3]
```diff
@@ -44,3 +44,57 @@ def test_excess_fields():
...
+    assert doc.expiration_date == '2030-12-31'
+    doc_no_fields = VerificationDocument(
+        country='MX',
+        region='',
+        photos=[],
+        steps=[],
+        type='national-id',
+        fields=None,
+    )
+    assert doc_no_fields.expiration_date == ''
```
[the remark this answers is not in the document]
Comment (rcabrera-py) [reply tests/test_types.py]: done

File: mati/types/enums.py (hunk 1 above)
Comment (rcabrera-py) [reply mati/types/enums.py]: done

File: tests/test_types.py [hunk 4]
```diff
@@ -44,3 +44,63 @@ def test_excess_fields():
...
+def test_expiration_date_property(fields, expected):
+    doc = VerificationDocument(
+        country='MX',
+        region='',
+        photos=[],
+        steps=[],
+        type='national-id',
+        fields=fields,
+    )
+    assert doc.expiration_date == expected
```
[the remark this answers is not in the document]
Comment (rcabrera-py) [reply tests/test_types.py:81]: Thanks for the detailed review! We've implemented the documentation suggestion:

✅ Applied: Added clear documentation explaining why date fields return None while other fields return empty strings (''). This documents the semantic difference: None indicates absence of a date value, while '' indicates an empty but present field.
❌ Not applied: We're keeping the -> None return type annotations for expiration_date and emission_date. After discussion with the team, we decided that since these properties can return None, the current annotation is appropriate for our use case.

The parametrized tests now cover all edge cases comprehensively. Thanks for the thorough code review!

[the remark this answers is not in the document]
Comment (rcabrera-py) [reply tests/test_types.py:81]: Thanks for the detailed feedback, CodeRabbit! You were absolutely right about the -> None annotation being factually incorrect for properties that can return a value other than None.
We've decided to update the return type annotations for expiration_date and emission_date to -> Optional[date]. This allows us to return actual datetime.date objects when a date is present and valid, and None when it's missing or invalid, providing a much clearer and more useful API.
We've also updated the documentation to reflect this change and explain the rationale. The linter and tests are now passing with this new implementation.
Thanks again for catching this!

File: tests/test_types.py [hunk 5]
```diff
@@ -44,3 +46,63 @@ def test_excess_fields():
...
+                }
+            },
+            date(2030, 12, 31),
+        ),
+        (None, None),
+        ({'address': {'value': 'Test Address', 'label': 'Address'}}, None),
+        ({'expiration_date': {'label': 'Date of Expiration'}}, None),
+    ],
+)
+def test_expiration_date_property(fields, expected):
```
Comment (felipao-mx) [inline tests/test_types.py]: en test también debemos agregar type annotations

File: tests/test_types.py [hunk 6]
```diff
@@ -44,3 +46,63 @@ def test_excess_fields():
...
+                    'value': '2030-12-31',
+                    'label': 'Date of Expiration',
+                    'format': 'date',
+                }
+            },
+            date(2030, 12, 31),
+        ),
+        (None, None),
+        ({'address': {'value': 'Test Address', 'label': 'Address'}}, None),
+        ({'expiration_date': {'label': 'Date of Expiration'}}, None),
```
Comment (felipao-mx) [inline tests/test_types.py:67]: duda, estos ejemplos de donde se obtuvieron? se tomaron de alguna prueba real?

Comment (rcabrera-py) [reply tests/test_types.py:67]: del archivo conftest.py y de los metadatos de metamap del OCR
image

File: tests/test_types.py (hunk 5 above)
Comment (rcabrera-py) [reply tests/test_types.py]: done

**Review by felipao-mx — APPROVED** [review]
